contributors/design-proposals/lifecycle-hooks.md

```diff
@@ -0,0 +1,152 @@
+# Lifecycle Hooks
```

Comment:
All docs in the design-proposals directly belong in SIG-specific subdirectories

Reply:
Yes, I'm gonna move it to KEP; this structure is year old based on what was there at the time it started.